frame/support/test/tests/storage_transaction.rs

```diff
@@ -210,3 +228,18 @@ fn transactional_annotation_in_decl_module() {
 		assert_noop!(<Module<Runtime>>::value_rollbacks(origin, 3), "nah");
 	});
 }
+
+#[test]
+#[should_panic(expected = "Require transaction not called within with_transaction")]
+fn require_transactional_panic() {
```

Comment:
This test will not panic on a local `cargo test --all --release` and that is bad.

I think the solution would be that we disable the test if `debug_assertions` are not enabled? Or do you got a better idea?

Reply:
test cfg check won't work for this so I think all we can do is disable it when `debug_assertions` is not enabled.

Reply:
In the end is that test here doing the same as the one from `frame-support` and we could just remove it.